eventstore_test.go

```diff
@@ -71,6 +80,92 @@ func TestNewEventConstructor(t *testing.T) {
 	is.Equal(e.Meta["foo"], "bar")
 }
 
+func TestEventStoreSnapshot(t *testing.T) {
+	is := testutil.NewIs(t)
+
+	srv := testutil.NewNatsServer(-1)
+	defer testutil.ShutdownNatsServer(srv)
+
+	nc, _ := nats.Connect(srv.ClientURL())
+
+	tr, err := types.NewRegistry(map[string]*types.Type{
+		"order-placed": {
+			Init: func() any { return &OrderPlaced{} },
+		},
+		"order-shipped": {
+			Init: func() any { return &OrderShipped{} },
+		},
+	})
+	is.NoErr(err)
+
+	r, err := New(t.Context(), nc, TypeRegistry(tr))
+	is.NoErr(err)
+
+	ctx := t.Context()
+
+	es, err := r.EventStore("orders", WithSnapshotSettings("snapshots"))
+	is.NoErr(err)
+	err = es.Create(&jetstream.StreamConfig{
+		Storage: jetstream.MemoryStorage,
+	})
+	is.NoErr(err)
+
+	events := []*Event{
+		{Data: &OrderPlaced{ID: "1"}},
+		{Data: &OrderPlaced{ID: "2"}},
+		{Data: &OrderPlaced{ID: "3"}},
+		{Data: &OrderShipped{ID: "2"}},
+	}
+
+	seq, err := es.Append(ctx, "orders.*", events)
+	is.NoErr(err)
+	is.Equal(seq, uint64(4))
+
+	var stats OrderStats
+	stats.Codec = codec.Default
+
+	seq2, err := es.Evolve(ctx, "orders.*", &stats, WithSnapshot("orderstats"))
+	is.NoErr(err)
+	is.Equal(seq, seq2)
+
+	is.Equal(stats.OrdersPlaced, 3)
+	is.Equal(stats.OrdersShipped, 1)
+
+	jsCtx, err := jetstream.New(nc)
+	is.NoErr(err)
+
+	jss, err := jsCtx.Stream(ctx, "orders")
+	is.NoErr(err)
+
+	err = jss.Purge(ctx)
+	is.NoErr(err)
+
+	events = []*Event{
+		{Data: &OrderPlaced{ID: "4"}},
+		{Data: &OrderShipped{ID: "1"}},
+	}
+
+	_, err = es.Append(ctx, "orders.*", events)
+	is.NoErr(err)
+
+	seq, err = es.Evolve(ctx, "orders.*", &stats)
+	is.NoErr(err)
+	is.Equal(stats.OrdersPlaced, 4)
+	is.Equal(stats.OrdersShipped, 2)
+	is.Equal(uint64(6), seq)
+
+	var loadStats OrderStats
+	loadStats.Codec = codec.Default
+
+	seqLoad, err := es.Evolve(ctx, "orders.*", &loadStats, FromSnapshot("orderstats", 0))
+	is.NoErr(err)
+	is.Equal(loadStats.OrdersPlaced, 4)
+	is.Equal(loadStats.OrdersShipped, 2)
+	is.Equal(uint64(6), seqLoad)
```

Comment:
I build this out to add events to the stream after the purge....found out that [this](https://github.com/synadia-labs/rita/pull/15/files#r2267967841) section was causing the count to get off.  Why are we skipping the first msg here??